### PR TITLE
Add items

### PR DESCRIPTION
**Pull request by oliverviljamaa**
*The animations are not working, not sure why yet.*

You can try it on: https://transferwise.github.io/public-navigation/branch/items

# v0.3.0
## Adds items

Items, originating from `src/items.json`, will be rendered as menu items, for mobile and desktop.

image

### CODE REVIEW

File: src/PublicNavigation.js [hunk 1]
```diff
@@ -1,7 +1,41 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { PropTypes as Types } from 'prop-types';
```
Comment (Tankenstein) [inline src/PublicNavigation.js]: Uhh does that work? Isn't it the default export?

File: src/PublicNavigationWithoutContent/Menu/Items/Items.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import Item from './Item';
+
+const Items = ({ items }) => (
+  <ul className="nav navbar-nav navbar-right">
+    {items.map(itemProps => <Item {...itemProps} key={itemProps.translationKey} />)}
+  </ul>
+);
+
+Items.propTypes = {
+  items: Types.arrayOf(Types.shape({})).isRequired,
```
Comment (Tankenstein) [inline src/PublicNavigationWithoutContent/Menu/Items/Items.js]: You should add `translationKey` as you are directly using it here.

File: src/PublicNavigationWithoutContent/Menu/Items/Item/Item.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { PropTypes as Types } from 'prop-types';
+
+const Item = ({ translationKey, link }) => (
```
Comment (OFFLlNE) [inline src/PublicNavigationWithoutContent/Menu/Items/Item/Item.js]: any specific reason to the order of arguments?

File: src/PublicNavigationWithoutContent/Menu/Items/Item/Item.spec.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+describe('Item', () => {
+  const item = shallow(<Item translationKey="business" link="https://transferwise.com/bisnes" />);
+
+  it('has translation key as link text', () => {
+    expect(text()).toBe('business');
+  });
+
+  it('has correct link', () => {
+    expect(link()).toBe('https://transferwise.com/bisnes');
```
Comment (OFFLlNE) [inline src/PublicNavigationWithoutContent/Menu/Items/Item/Item.spec.js]: 😂

**Comment by Tankenstein** [comment]
Animations are probably not working because they rely on the sidebar actually being in the dom, just hidden, so they can transition. For transitioning from actually being missing in the dom, you would need another solution, like react-css-transition-group.

**Comment by oliverviljamaa** [comment]
@Tankenstein I thought about that, but actually the whole DOM is there the whole time, we just shift the menu on the screen from off-the-screen when toggled.

File: src/PublicNavigation.js (hunk 1 above)
Comment (oliverviljamaa) [reply src/PublicNavigation.js]: It does work, saw errors in both browser console and tests as well.
`import Types from 'prop-types';` is nicer, though, so will switch to that.

File: src/PublicNavigationWithoutContent/Menu/Items/index.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import Items from './Items';
+
+export default Items;
```
Comment (codewhisperer) [inline src/PublicNavigationWithoutContent/Menu/Items/index.js]: Why is this file necessary? Just to make the import statements shorter?

File: src/PublicNavigationWithoutContent/Menu/Items/Item/Item.js (hunk 3 above)
Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/Menu/Items/Item/Item.js]: I guess I considered the key a bit more important, even though they appear in the reverse order in the DOM.

File: src/PublicNavigationWithoutContent/Menu/Menu.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { PropTypes as Types } from 'prop-types';
+
+import Items from './Items';
+import './Menu.less';
```
Comment (codewhisperer) [inline src/PublicNavigationWithoutContent/Menu/Menu.js]: Should we make `less` imports be bottom of the import section?

File: src/PublicNavigationWithoutContent/Menu/Menu.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import MenuToggle from '../common/MenuToggle';
+
+const className = 'tw-public-navigation__menu';
+
+const Menu = ({ items, isOpen, onToggle }) => (
+  <nav
+    id="navbar"
+    className={`${className} collapse navbar-collapse navbar-collapse-with-panel${
+      isOpen ? ' in' : ''
```
Comment (codewhisperer) [inline src/PublicNavigationWithoutContent/Menu/Menu.js]: There is a lib called classnames(https://www.npmjs.com/package/classnames) that solves this prettier I think

File: src/PublicNavigationWithoutContent/Menu/Items/index.js (hunk 5 above)
Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/Menu/Items/index.js]: Yes, that's the convention normally, right?

File: src/PublicNavigationWithoutContent/Menu/Menu.js (hunk 6 above)
Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/Menu/Menu.js]: Yeah, that'd actually be nicer.

File: src/PublicNavigationWithoutContent/Menu/Menu.js (hunk 7 above)
Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/Menu/Menu.js]: I'd be happy to use it. @Tankenstein Do we use it elsewhere and what do you think?

Comment (Tankenstein) [reply src/PublicNavigationWithoutContent/Menu/Menu.js]: I don't think we're using it in libs, but are using it in some apps. I'd not add this here, as it's just another peerdependency for little gain.

Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/Menu/Menu.js]: Cool, I'll keep it in mind and when it's getting too complex, will reconsider.

File: src/PublicNavigationWithoutContent/Menu/Menu.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    }`}
+  >
+    <MenuToggle
+      isMenuOpen={isOpen}
+      onToggle={onToggle}
+      className={`cover${isOpen ? '' : ' collapsed'}`}
```
Comment (codewhisperer) [inline src/PublicNavigationWithoutContent/Menu/Menu.js]: Isn't it more logical to add the space before the `$` instead of in the return value?

File: src/PublicNavigationWithoutContent/PublicNavigationWithoutContent.js [hunk 9]
```diff
@@ -1,24 +1,41 @@
...
 PublicNavigationWithoutContent.propTypes = {
+  items: Types.arrayOf(Types.shape()),
   inverse: Types.bool,
   logoLink: Types.string,
+  isMenuOpen: Types.bool,
+  onToggleMenu: Types.func,
 };
 
 PublicNavigationWithoutContent.defaultProps = {
+  items: [],
```
Comment (codewhisperer) [inline src/PublicNavigationWithoutContent/PublicNavigationWithoutContent.js]: Shouldn't items be `required` ?

File: src/PublicNavigationWithoutContent/Menu/Menu.js (hunk 8 above)
Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/Menu/Menu.js]: I don't like it as then it would be `class="cover "`, not `class="cover"` 🤔

File: src/PublicNavigationWithoutContent/PublicNavigationWithoutContent.js (hunk 9 above)
Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/PublicNavigationWithoutContent.js]: Yeah, I think they should.

File: src/PublicNavigationWithoutContent/common/MenuToggle/MenuToggle.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+MenuToggle.propTypes = {
+  isMenuOpen: Types.bool.isRequired,
+  onToggle: Types.func.isRequired,
+  className: Types.string,
+  children: Types.node,
+};
+
+MenuToggle.defaultProps = {
+  className: '',
+  children: null,
```
Comment (codewhisperer) [inline src/PublicNavigationWithoutContent/common/MenuToggle/MenuToggle.js]: Is this necessary? 
Wouldn't it be ok if children were undefined?

Comment (oliverviljamaa) [reply src/PublicNavigationWithoutContent/common/MenuToggle/MenuToggle.js]: Why do you think `children: undefined` is better here than `children: null`?

Comment (codewhisperer) [reply src/PublicNavigationWithoutContent/common/MenuToggle/MenuToggle.js]: Not better but the same and saves a line of code

Comment (codewhisperer) [reply src/PublicNavigationWithoutContent/common/MenuToggle/MenuToggle.js]: Ah ok, eslint wants something there so can't save a line of code ☹️

**Comment by oliverviljamaa** [comment]
oki

**Comment by oliverviljamaa** [comment]
oki

**Comment by oliverviljamaa** [comment]
oki

**Comment by oliverviljamaa** [comment]
oki
